pkg/image/registry/image/strategy.go

```diff
@@ -74,6 +74,15 @@ func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newImage.DockerImageMetadataVersion = oldImage.DockerImageMetadataVersion
 	newImage.DockerImageLayers = oldImage.DockerImageLayers
 	newImage.Signatures = oldImage.Signatures
+	newImage.DockerImageConfig = oldImage.DockerImageConfig
```

Comment:
since we allow to change manifest, we should allow to change docker image config the same way if its digest matches the one in the manifest